### ISSUE TITLE
Impossible to create/sync tokens into variable in multiple Figma files keeping cross references.

### ISSUE THREAD

**Issue by FossilsCompany** [tokens-studio/figma-plugin#2149]
**Describe the bug**
This issue is related to the one closed a few weeks ago : #2055.

I was in vacation when Six asked me more detailed about this issue, so I wasn't there to answer, sorry for that.

I answered in the #2055 thread to explain my issue. Hope it will be sufficient. 

Briefly, yes I'm agree it's possible to create and synchronize variables with one Figma file. But as I explained I'm working with 3 of them (for the moment). One holding our Core variables, one for our semantic variables and the last one to create all the components. Except the core figma files, semantic and components figma files refer to variables from the core and semantic figma files.

So what I need is the possibility to keep the references is each of my figma files at any time. It tooks me hours to recreate all figma references. And more than that the possibility to be always in sync with Token Studio. As I said exept if I'm wrong (and frankly I would love it) it is not possible today to sync anything without loosing all my figma references. They will be replaced by references to variable created in the same file.

On my perspective, and this one is limited obviously, either it's possible to store file location of a variable in Token Studio. Meaning if I click on create/sync variable Token studio can recreate the proper reference. Or and it's a minimum, if I tag a set in token studio as a source, then all related token are marked as "not published" automatically.

Hope it is clearer... :)

Tks to you.

**Comment by rbosker** [comment]
@FossilsCompany Can you check this Feature Request #2023? I believe that describes the issue you're currently facing.

**Comment by mynikki04** [comment]
Do I need to get a Pro Plan to use Sync and Create variable features? The option has been disabled.

**Comment by keeganedwin** [comment]
Hey @mynikki04, that is correct if you aren't a **Pro** user as you require it to create **Themes**, which in turn are required to create variables.

**Comment by esthercheran** [comment]
@FossilsCompany I believe #1974 and #2023 will solve your use case. Let me know if it covers your use case.

**Comment by XavierVilars** [comment]
@esthercheran yes it would be as soon as it is possible to sync with mutliple Figma files.

**Comment by UdayHyma** [comment]
Added to Featurebase Roadmap : [❖ ❇️ Non-local support of variables and styles
](https://tokensstudio.featurebase.app/p/non-local-support-of-variables)
